lib/src/builder/parsing/members/props_and_state_util.dart

```diff
@@ -62,6 +64,18 @@ annotations.TypedMap getPropsOrStateAnnotation(bool isProps, AnnotatedNode node)
           // ignore: deprecated_member_use_from_same_package
           InstantiatedMeta.fromNode<annotations.StateMixin>(node));
 
+  if (meta != null && meta.potentiallyIncompleteValue is annotations.Props) {
+    // fixme error when this case isn't well formed
+    if (meta.unsupportedArguments.length == 1) {
+      final arg = meta.unsupportedArguments[0];
+      if (arg is NamedExpression && arg.name.label.name == 'ignoreRequiredProps') {
+        props_ignoreRequiredProps_source[meta.potentiallyIncompleteValue] =
```

Comment:
Ooh I initially used this expando solution because there wasn't a way to instantiate the `Props` with real references to the types for `ignoreRequiredPropsFrom`, and was pretty unhappy with how sketchy it was. E.g., if we ended up with a different instance of props somehow, then we'd lose that value.

Ideally I'd love to use parse the expression and pass that data into the real `Props` object so we can eliminate that expando, if it's not too much more effort.

Something like this:
```dart
      if (arg is NamedExpression && arg.name.label.name == 'ignoreRequiredProps') {
        // Attempt to parse the value, and fall through if something goes wrong,
        // and let `meta?.value` below throw.
        final expression = arg.expression;
        if (expression is SetOrMapLiteral && expression.isSet) {
          final simpleStringElements =
              expression.elements.whereType<SimpleStringLiteral>().toList();
          if (simpleStringElements.length == expression.elements.length) {
            return annotations.Props(
              keyNamespace: meta.potentiallyIncompleteValue.keyNamespace,
              ignoreRequiredProps: simpleStringElements.map((e) => e.value).toSet(),
            );
          }
        }
      }
```

I'd be fine doing this as a follow-up ticket, too